src/coredump/coredump.c

```diff
@@ -1560,8 +1626,69 @@ static int forward_coredump_to_container(Context *context) {
         return 0;
 }
 
+static int gather_pid_mount_tree_fd(const Context *context) {
+        _cleanup_close_ int mntns_fd = -EBADF, root_fd = -EBADF;
+        _cleanup_close_pair_ int pair[2] = EBADF_PAIR;
+        int fd = -EBADF, r;
+        pid_t child;
+
+        assert(context);
+
+        /* Don't bother preparing environment if we can't pass it to libdwfl. */
+#if !HAVE_DWFL_SET_SYSROOT
+        return -EBADF;
```

Comment:
Why EBADF? EOPNOTSUPP?

Reply:
Function is supposed to prepare mount tree fd, if it fails to do that or there is no point in doing that it returns invalid fd, I think return code makes sense.

Reply:
"invalid fd" could mean anything. Let's be accurate here.

Reply:
it needs to be EBADF as the return value is assigned to the fd var, as it is commonly done - ie, this function returns an fd, not just an error code

Reply:
It doesn't _need_ EBADF. All negative fds are valid. E.g. look at getpeerpidfd, which is honest about error and all our code can cope with that

Reply:
Btw, we are honest about errors, note that -EBADF is used to distinguish when to log about the errors. So -EBADF means we didn't get a proper fd but it is fine and we don't log and continue. Hence, when container access is disabled or we don't have library support then we didn't get fd but it fine and expected.